First draft workshop content (#3)
observability-lab-cse/observability-lab-workshop: 01-provision-infrastructure/README.md

> Goal: Achieving a fully operational infrastructure.

<!-- - Utilize Bash and Bicep scripts for infrastructure installation containing: AKS, AppInsighrs, Log Analyrtics workspace, etc.

> - Quick run through of ech resource and its uses for the workshop -->

Reviewer: ```suggestion
- Quick run through of each resource and its uses for the workshop -->
```

## 1. Architecture

> First, let's take a closer look at the application we'll be using for this workshop. It's a XXXX solution comprising two microservices and a set of Azure resources. While this solution is rather simplistic and not a real-world example, its purpose is to provide you with hands-on experience in instrumenting various services written in different languages. This way, you'll become more comfortable with instrumenting your own solutions. Below, you'll find a diagram outlining our solution and the required resources

Reviewer: Maybe we can add some details of what the application does? Something like:

```suggestion
First, let's take a closer look at the application we'll be using for this workshop. It's a solution comprising two microservices, one written in C# and the other in Java, and a set of Azure resources. The solution allows users to onboard and manage smart-home devices. The application generates sample telemetry (e.g. temperature values) for the provisioned devices. While this solution is rather simplistic and not a real-world example, its purpose is to provide you with hands-on experience in instrumenting various services written in different languages. This way, you'll become more comfortable with instrumenting your own solutions. Below, you'll find a diagram outlining our solution and the required resources
```

![Architecture](solution-resources.jpg)

## 2. Provisioning

Before we dive into the workshop, it's essential to have your Azure resources prepared. Think of this as laying the foundation for our upcoming steps.

To simplify this process for everyone, we have provided Bicep and Bash scripts that can create the entire infrastructure in one go.

Here's what you need to do: Run the following command in the root of the repository, take a coffee break, and when you return, your resources should be provisioned and ready to go ☕️.

> 

Reviewer: I think we should also mention the .env file and that some variables need to be populated, specifically `ENV_RESOURCE_GROUP_NAME` and `ENV_PROJECT_NAME`.

Reviewer: We have a section about this in the README of the 'full' repo:

The Makefile automates infrastructure creation and applications deployment within a couple of commands. To use it create a .env in the root of the project and add the following content to it.

ENV_RESOURCE_GROUP_NAME=<rg-name>
ENV_LOCATION="westeurope"
ENV_PROJECT_NAME=<project-name> # should just be lowercase letters or numbers

Author: Thats a good note!

```sh
az login
az account set --subscription <name or ID of your subscription>

make provision
```

> > Note: If you are interested in how its deployed and all the bicep scripts. Dont worry, here they all are: [/infra](https://github.com/observability-lab-cse/observability-lab/tree/section/01-provision-infrastructure/infrastructure)

Reviewer: ```suggestion
> Note: If you are interested in how its deployed and all the bicep scripts. Dont worry, here they all are: [/infrastructure](https://github.com/observability-lab-cse/observability-lab/tree/section/01-provision-infrastructure/infrastructure)
```

Running the script will provision the following essential resources for your application:

- AKS (Azure Kubernetes Service): This is where your applications will be hosted.
- CosmosDB: This serves as your device registry.
- KeyVault: A secure place to store your application secrets.
- EventHub: For real-time event streaming or in our case used to send and receive device temperature and health data
- Storage Account: This component is essential for the secure storage of EventHub checkpoints

> - ACR (Azure Container Registry): For the application conatiner images.

Reviewer: ```suggestion
- ACR (Azure Container Registry): To store the application container images.
```
